mathesar/api/display_options.py

```diff
@@ -1,34 +1,70 @@
+            {
+                "name": "currency_details",
+                "type": "object",
+                "items": [
+                    {"name": "symbol", "type": "string"},
+                    {"name": "symbol_location", "type": "number", "enum": [1, -1]},
+                    {"name": "decimal_symbol", "type": "string", "enum": [",", "."]},
+                    {"name": "digit_grouping", "type": "array"},
+                    {"name": "digit_symbol", "type": "string", "enum": [",", ".", " "]}
+                ]
+            }]
+    }
+
+
 DISPLAY_OPTIONS_BY_TYPE_IDENTIFIER = {
-    MathesarTypeIdentifier.BOOLEAN.value:
-        {
-            "options": [{"name": "input", "type": "string",
-                         "enum": ['dropdown', 'checkbox']},
-                        {'name': "custom_labels", "type": "object",
-                         "items": [{"name": "TRUE", "type": "string"},
-                                   {'name': "FALSE", "type": "string"}]}]
-
-        },
-    MathesarTypeIdentifier.NUMBER.value:
-        {
-            "options": [{"name": "show_as_percentage", "type": "boolean"},
-                        {"name": "locale", "type": "string"}]
-        },
-    MathesarTypeIdentifier.DATETIME.value:
-        {
-            "options": [{"name": "format", "type": "string"}]
-        },
-    MathesarTypeIdentifier.TIME.value:
-        {
-            "options": [{"name": "format", "type": "string"}]
-        },
-    MathesarTypeIdentifier.DATE.value:
-        {
-            "options": [{"name": "format", "type": "string"}]
-        },
-    MathesarTypeIdentifier.DURATION.value:
-        {
-            "options": [{"name": "format", "type": "string"}]
-        }
+    MathesarTypeIdentifier.BOOLEAN.value: lambda: {
+        "options": [
+            {
+                "name": "input", "type": "string",
+                "enum": ['dropdown', 'checkbox']
+            },
+            {
+                'name': "custom_labels", "type": "object",
+                "items": [
+                    {"name": "TRUE", "type": "string"},
+                    {'name': "FALSE", "type": "string"}
+                ]
+            }
+        ]
+
+    },
+    MathesarTypeIdentifier.DATE.value: lambda: {"options": [{"name": "format", "type": "string"}]},
+    MathesarTypeIdentifier.DATETIME.value: lambda: {"options": [{"name": "format", "type": "string"}]},
+    MathesarTypeIdentifier.DURATION.value: lambda: {"options": [{"name": "format", "type": "string"}]},
+    MathesarTypeIdentifier.MONEY.value: money_display_options_schema,
+    MathesarTypeIdentifier.NUMBER.value: lambda: {
+        "options": [
+            {
+                "name": "show_as_percentage",
+                "type": "string",
+                "enum": ['dropdown', 'checkbox']
+            },
+            {
+                "name": "locale",
+                "type": "string"
+            }
+        ]
```

Comment:
I realize this is carried over, but I think we should change this to be more like the money display options, as I mentioned in a comment elsewhere. There are plenty of reasons to display numbers using configurations not represented by a locale. It has to do with preferences, not just your location.